core/templates/pages/exploration-player-page/services/checkpoint-celebration-utility.service.spec.ts

```diff
@@ -0,0 +1,235 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the checkpoint celebration utility service.
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { EventEmitter, NO_ERRORS_SCHEMA } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+
+import { CheckpointCelebrationUtilityService } from 'pages/exploration-player-page/services/checkpoint-celebration-utility.service';
+import { ComputeGraphService } from 'services/compute-graph.service';
+import { StateObjectsBackendDict } from 'domain/exploration/StatesObjectFactory';
+import { StatesObjectFactory } from 'domain/exploration/StatesObjectFactory';
+
+class MockTranslateService {
+  onLangChange: EventEmitter<string> = new EventEmitter();
+  instant(key: string, interpolateParams?: Object): string {
```

Comment:
Why `?`?

Reply:
That input is optional. It is provided when a string needs to be interpolated into a translation string. If the translation string doesn't isn't written to support interpolation, this value isn't provided in the function call.

Reply:
Hmm, in that case do not use `?` but mention `undefined` explicitly in the type. Also in that case cannot we rather pass empty object?

Reply:
Done. As for passing an empty object, I didn't do that because in most (if not all) other instances of obtaining translations using this service/method, if the string doesn't support interpolation the `interpolateParams` value is simply not supplied.